### PR TITLE
WIP Poly mask

### PR DESCRIPTION
**Pull request by aarora8**
adding changes for mask generation, obtaining point from a polygon, validation scripts.

### CODE REVIEW

**Comment by danpovey** [comment]
I merged this PR in the interest of keeping thing moving fast.  Thanks, Ashish!
Yiwen, I hope that you can take the lead in modifying your training code to use these utilities. 
If you want you can create another version of your training script that will use this, since for now it won't work.  This will help us get this stuff functional as soon as possible.

**Comment by YiwenShaoStephen** [comment]
OK, I will check it and make modifications accordingly.
